include/glow/Graph/Graph.h

```diff
@@ -183,6 +183,7 @@ class Function final : public Named {
   /// Inserts the node \p N to the list of nodes, and returns the inserted node.
   template <class NodeTy> NodeTy *addNode(NodeTy *N) {
     N->setName(Module::uniqueName(N->getName(), uniqueNodeNames_));
+    N->setParent(this);
```

Comment:
Besides adding the nodes, we also need to handle the removal of nodes and set the parent to nullptr. And we need to the corresponding check to the test you introduced.

Reply:
What do you mean?
We don't have a removeNode API, do we?

Reply:
We have `eraseNode`

Reply:
And may be we should have `removeNode` as well.